### PR TITLE
[CHNL-13250] dismiss webview through form actions

### PR DESCRIPTION
**Pull request by ab1470**
# Description

This PR adds a `webViewDidClose()` delegate method to detect when the form emits a `window.close()` javascript event. The method will call the ViewModel's `dismiss()` method, which will be handled after we've wired up the SDK to display the IAM view in the host app.

# Check List

- [ ] Are you changing anything with the public API?
- [ ] Have you tested this change on real device?
- [x] Are your changes backwards compatible with previous SDK Versions?
- [ ] Have you added unit test coverage for your changes?
- [x] Have you verified that your changes are compatible with all the operating system version this SDK currently supports?

# Supporting Materials

Here's a video showing a message being printed to the Xcode console when the user taps a "Close" button:

https://github.com/user-attachments/assets/23e71442-3a42-403a-8fff-203a025b3b51

### CODE REVIEW

File: Sources/KlaviyoUI/KlaviyoWebView/KlaviyoWebViewController.swift [hunk 1]
```diff
@@ -91,6 +91,10 @@ class KlaviyoWebViewController: UIViewController, WKUIDelegate {
         }
     }
 
+    func webViewDidClose(_ webView: WKWebView) {
```
Comment (ndurell) [inline Sources/KlaviyoUI/KlaviyoWebView/KlaviyoWebViewController.swift:94]: So this is called by js?

File: Sources/KlaviyoUI/KlaviyoWebView/KlaviyoWebWrapperViewController.swift [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        loadSubviews()
+    }
+
+    func loadSubviews() {
+        if let blurEffectView {
+            view.addSubview(blurEffectView)
+            blurEffectView.pin(to: view)
+        }
+
+        if let tintView {
```
Comment (ndurell) [inline Sources/KlaviyoUI/KlaviyoWebView/KlaviyoWebWrapperViewController.swift:98]: Is this the new way to check for nil?

**Review by ndurell — APPROVED** [review]

File: Sources/KlaviyoUI/KlaviyoWebView/KlaviyoWebWrapperViewController.swift (hunk 2 above)
Comment (ab1470) [reply Sources/KlaviyoUI/KlaviyoWebView/KlaviyoWebWrapperViewController.swift:98]: yeah, no need to do `if let tintView = tintView` anymore

File: Sources/KlaviyoUI/KlaviyoWebView/KlaviyoWebViewController.swift (hunk 1 above)
Comment (ab1470) [reply Sources/KlaviyoUI/KlaviyoWebView/KlaviyoWebViewController.swift:94]: out of the box, WKWebView will listen for a `window.close()` JS event. We can respond to the close event by implementing this delegate method

**Comment by ab1470** [comment]
Closing this as we've decided on a different approach for MVP. We can revisit this at a future date, if necessary.

**Pull request closed without merging**